learn/features/randomness.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Randomness
+description: Learn about the sources of VRF randomness on Moonbeam, the request and fulfillment process, and some security considerations when using on-chain randomness.
+---
+
+# Randomness on Moonbeam
+
+![Randomness Moonbeam Banner](/images/learn/features/randomness/randomness-banner.png)
+
+## Introduction {: #introduction } 
+
+Randomness is necessary for a variety of blockchain applications to create unbiased, unpredictable, and unique outcomes. However, obtaining a reliable source of randomness is a challenge. Computers are deterministic, meaning given the same input, the same output will always be produced. Therefore, random values generated by computers are referred to as pseudo-random as they appear to be statistically random, but given the same input, the output can easily be repeated. 
+
+Moonbeam utilizes verifiable random functions (VRF) to generate randomness that can be verified on-chain. A VRF is a cryptographic function that takes some input and produces random values, along with a proof of authenticity that these random values were generated by the submitter. The proof can be verified by anyone to ensure the random values generated were calculated correctly.
+
+There are two available sources of randomness that provide random inputs based on block producers' VRF keys and past randomness results: [local VRF](#local-vrf) and [BABE epoch randomness](#babe-epoch-randomness). Local VRF is determined directly within Moonbeam using the collator of the block's VRF key and the last block's VRF output. On the other hand, [BABE](https://wiki.polkadot.network/docs/learn-consensus#block-production-babe){target=_blank} epoch randomness is based on all the VRF produced by the relay chain validators during a complete [epoch](https://wiki.polkadot.network/docs/glossary#epoch){target=_blank}.
+
+You can interact with and request on-chain randomness using the randomness precompile contract, a Solidity interface that enables smart contract developers to access the randomness functionality through the Ethereum API. For more information, please check out the [Interacting with the Randomness Precompile](/builders/pallets-precompiles/precompiles/randomness){target=_blank} guide. You can also take a look at the [The Randomness Pallet](/builders/pallets-precompiles/pallets/randomness){target=_blank}, which can be used to obtain current randomness requests and results.
+
+## General Definitions {: #general-definitions } 
+
+- **Epoch** - a time duration in the BABE protocol that is broken into smaller time slots. Slots are discrete units of time six seconds in length. On Polkadot, one epoch lasts approximately 2,400 slots or 4 hours. On Kusama, one epoch lasts approximately 600 slots or 1 hour.
```

Comment:
We could leave this values as constant as they don't normally change right?

Reply:
yeah that was my thinking 🤷‍♀️